File: src/main/java/com/kaishiyoku/youtubedownloadhelper/Channel.java
package com.kaishiyoku.youtubedownloadhelper;

public class Channel {
    private String description;
    private String url;
    private String localPath;

    public Channel(String description, String url, String localPath) {
        this.description = description;
        this.url = url;
        this.localPath = localPath;
    }

    public Channel(String configLine) {
        String[] values = configLine.split(";");

        description = values[0];
        url = values[1];
        localPath = values[2];
    }

    public String getDescription() {
        return description;
    }

    public String getUrl() {
        return url;
    }

    public String getLocalPath() {
        return localPath;
    }
}

File: src/main/java/com/kaishiyoku/youtubedownloadhelper/models/Channel.kt
package com.kaishiyoku.youtubedownloadhelper.models

import com.google.gson.Gson
import com.kaishiyoku.youtubedownloadhelper.helper.ConsoleHelper.defaultGson

class Channel(description: String, url: String, localPath: String) {
    var description: String = description
        private set
    var url: String = url
        private set
    var localPath: String = localPath
        private set

    companion object {
        fun fromConfigLine(configLine: String): Channel {
            val (description, url, localPath) = configLine.split(";").dropLastWhile { it.isEmpty() }.toTypedArray()

            return Channel(description, url, localPath)
        }

        fun fromJson(json: String) : Channel {
            return defaultGson().fromJson(json, Channel::class.java)
        }
    }
}
